lib/util.js

```diff
@@ -0,0 +1,69 @@
+const ApiError = require('./error');
+
+/**
+ * Automatically retry a request if it fails with an appropriate status code.
+ *
+ * A GET request is retried if it fails with a 429 or 5xx status code.
+ * A non-GET request is retried only if it fails with a 429 status code.
+ *
+ * If the response sets a Retry-After header,
+ * the request is retried after the number of seconds specified in the header.
+ * Otherwise, the request is retried after the specified interval,
+ * with exponential backoff and jitter.
+ *
+ * @param {Function} request - A function that returns a Promise that resolves with a Response object
+ * @param {object} options
+ * @param {Function} [options.shouldRetry] - A function that returns true if the request should be retried
+ * @param {number} [options.maxRetries] - Maximum number of retries. Defaults to 5
+ * @param {number} [options.interval] - Interval between retries in milliseconds. Defaults to 500
+ * @returns {Promise<Response>} - Resolves with the response object
+ * @throws {ApiError} If the request failed
+ */
+async function withAutomaticRetries(request, options = {}) {
```

Comment:
I really wanted this function to take the `fetch` function and a real `Request` object. However, `node-fetch` has a bug where passing a `Request` object into `fetch` throws `TypeError: Only absolute URLs are supported`. The error has to do with incorrect `isRequest` method, or else broken URL normalization in `getNodeRequestOptions`.

To work around this, I'm passing `shouldRetry` as a closure.